cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -567,6 +579,7 @@ func (scaleSet *ScaleSet) setInstanceStatusByProviderID(providerID string, statu
 			scaleSet.instanceCache[k].Status = &status
 		}
 	}
+	scaleSet.lastInstanceRefresh = time.Now()
```

Comment:
This added recently with the same motivation too. If we proactively update the instance state to `Deletion`, we don't want it to be invalidated in the next loop in case that cache is stale.